packages/noir-ethereum-api/test/e2e.test.ts

```diff
@@ -22,39 +21,45 @@ const defaultTestCircuitInputParams: MainInputs = {
 
 describe(
   'e2e',
-  async () => {
-    async function oracles(accountWithProof: AccountWithProof = accountWithProofJSON): Promise<Oracles> {
-      return createOracles(await createMockClient('./test/fixtures/mockClientData.json'))({
-        get_account: async () => serializeAccountWithProof(accountWithProof),
-        get_header: async () => encodeBlockHeaderPartial(blockHeaders[1].header as BlockHeader)
-      });
-    }
+  () => {
+    let client: PublicClient;
+    let oracles: Oracles;
+
+    beforeAll(async () => {
+      client = await createMockClient('./test/fixtures/mockClientData.json');
+      oracles = createOracles(client)(defaultOraclesMap);
+    });
 
     it('proof successes', async () => {
-      expect(await generateAndVerifyStorageProof(defaultTestCircuitInputParams, await oracles())).toEqual(true);
+      expect(await generateAndVerifyStorageProof(defaultTestCircuitInputParams, oracles)).toEqual(true);
     });
 
-    const arrayKeys: Array<FieldsOfType<AccountWithProof, readonly string[]>> = ['key', 'value', 'proof'];
-    arrayKeys.forEach((arrayField) => {
-      it(`proof fails: invalid field: ${arrayField}`, async () => {
-        await expectCircuitFail(
-          generateAndVerifyStorageProof(
-            defaultTestCircuitInputParams,
-            await oracles({
-              ...accountWithProofJSON,
-              [arrayField]: alterArray(accountWithProofJSON[arrayField])
-            })
-          )
-        );
-      });
+    it('proof fails: invalid proof', async () => {
+      const resultModifier = (call: Call): Call => {
+        if (call.method === 'getProof') {
+          const accountProof = (call.result as { [key: string]: object }).accountProof as string[];
```

Comment:
TODO: implement util method for updating object so that we could use:
```ts
update(call, 'result.accountProof.0', incHexStr)
```

Reply:
We can copy a 6-liner from here:
https://youmightnotneed.com/lodash#set